engine/src/flutter/display_list/dl_op_records.h

```diff
@@ -1001,35 +1000,24 @@ struct DrawDisplayListOp final : DrawOpBase {
 
 // 4 byte header + 8 payload bytes + an aligned pointer take 24 bytes
 // (4 unused to align the pointer)
-struct DrawTextBlobOp final : DrawOpBase {
-  static constexpr auto kType = DisplayListOpType::kDrawTextBlob;
+struct DrawTextOp final : DrawOpBase {
+  static constexpr auto kType = DisplayListOpType::kDrawText;
 
-  DrawTextBlobOp(const sk_sp<SkTextBlob>& blob, DlScalar x, DlScalar y)
-      : DrawOpBase(kType), x(x), y(y), blob(blob) {}
+  DrawTextOp(const std::shared_ptr<DlText>& text, DlScalar x, DlScalar y)
+      : DrawOpBase(kType), x(x), y(y), text(text) {}
 
   const DlScalar x;
   const DlScalar y;
-  const sk_sp<SkTextBlob> blob;
-
-  void dispatch(DlOpReceiver& receiver) const {
-    receiver.drawTextBlob(blob, x, y);
-  }
-};
-
-struct DrawTextFrameOp final : DrawOpBase {
-  static constexpr auto kType = DisplayListOpType::kDrawTextFrame;
+  const std::shared_ptr<DlText> text;
 
-  DrawTextFrameOp(const std::shared_ptr<impeller::TextFrame>& text_frame,
-                  DlScalar x,
-                  DlScalar y)
-      : DrawOpBase(kType), x(x), y(y), text_frame(text_frame) {}
+  void dispatch(DlOpReceiver& receiver) const { receiver.drawText(text, x, y); }
 
-  const DlScalar x;
-  const DlScalar y;
-  const std::shared_ptr<impeller::TextFrame> text_frame;
-
-  void dispatch(DlOpReceiver& receiver) const {
-    receiver.drawTextFrame(text_frame, x, y);
+  DisplayListCompare equals(const DrawTextOp* other) const {
+    return text->getTextBlob() == other->text->getTextBlob() &&
+                   text->getTextFrame() == other->text->getTextFrame() &&
```

Comment:
Perhaps put this pointer comparison inside a `DlText::==` operator to hide the details (and in case we eventually have some better way to compare them than by (pair of) subsumed pointer(s)?)